### DISCUSSION TITLE
I want to contribute.

### DISCUSSION THREAD

**Discussion by JAY-TRIVEDI-007** [makeplane/plane#4403] [Q&A]
Hello,
I want to contribute to this project. I read the contributing guidelines.
Please assign me a bug issue to start from.
The question is if I pick a bug and work on it and if before I am done fixing it someone already fixed it.
How does that work in open-source projects?
Thank you.

**Comment by nyxb** [comment]
> Hello,
> I want to contribute to this project. I read the contributing guidelines.
> Please assign me a bug issue to start from.
> The question is if I pick a bug and work on it and if before I am done fixing it someone already fixed it.
> How does that work in open-source projects?
> Thank you.

In open-source projects, it’s common for multiple contributors to show interest in fixing the same issue. To avoid duplication of effort, most projects have some guidelines in place:

Claiming the Issue: Typically, once you comment on an issue and express interest in working on it, maintainers will assign the issue to you (or acknowledge that you're working on it).

Communication: Keep communicating your progress in the issue thread, so others know you're actively working on it. If you’re delayed, let the maintainers know.

First-Come, First-Served: In case someone else submits a fix before you do, most projects will accept the first valid pull request. If someone else beats you to it, your fix might not be merged, but you can still learn from the experience.

The best way to handle this is by staying in touch with the project maintainers and other contributors to ensure smooth collaboration!

**Reply by JAY-TRIVEDI-007** [reply]
Thank you for your explanation
